mmdet3d/core/evaluation/seg_eval.py

```diff
@@ -66,28 +66,44 @@ def get_acc_cls(hist):
     return np.nanmean(np.diag(hist) / hist.sum(axis=1))
 
 
-def seg_eval(gt_labels, seg_preds, label2cat, logger=None):
-    """Semantic Segmentation Evaluation.
+def seg_eval(gt_labels,
+             seg_preds,
+             label2cat,
+             num_classes,
+             ignore_index,
+             logger=None):
+    """Semantic Segmentation  Evaluation.
 
      Evaluate the result of the Semantic Segmentation.
 
      Args:
-         gt_labels (list[torch.Tensor]): Ground truth labels.
-         seg_preds  (list[torch.Tensor]): Predtictions
-         label2cat (dict): Map from label to category.
-         logger (logging.Logger | str | None): The way to print the mAP
+        gt_labels (list[torch.Tensor]): Ground truth labels.
+        seg_preds  (list[torch.Tensor]): Predtictions
+        num_classes (int): Number of categories.
+        label2cat (dict): Map from label to category.
+        ignore_index (int): Index that will be ignored in evaluation.
+        logger (logging.Logger | str | None): The way to print the mAP
             summary. See `mmdet.utils.print_log()` for details. Default: None.
 
-    Return:
+    Returns:
         dict[str, float]: Dict of results.
     """
     assert len(seg_preds) == len(gt_labels)
 
+    # filter out ignored points
+    for i in range(len(gt_labels)):
+        gt_seg = gt_labels[i].clone()
+        pred_seg = seg_preds[i].clone()
```

Comment:
The cloning of gt is indeed unnecessary. The cloning of pred is necessary because I modify some values and this will affect the variable outside seg_eval function.

Reply:
But `pred_seg` is still modified?